src/services/notifications/api.ts

```diff
@@ -0,0 +1,115 @@
+import firestore from '@react-native-firebase/firestore';
+import { DocumentData, DocumentRefData } from '../../types/firebase';
+
+import userAPI from '../user/api';
+
+export enum NotificationType {
+  standard = 'STD',
+  adoptionInterest = 'AIN',
+  adoptionRefused = 'ARE',
+}
+
+export interface INotificationAdoptionInterestData {
+  animal: DocumentRefData,
+}
+
+export interface NotificationModel {
+  id: string,
+  from: DocumentData,
+  to: DocumentRefData,
+  message: string,
+  seen: false,
+}
+
+export interface NotificationStandardModel extends NotificationModel {
+  type: NotificationType.standard,
+}
+
+export interface NotificationAdoptionModel extends NotificationModel {
+  type: NotificationType.adoptionInterest,
+  animal: DocumentData,
+}
+
+export interface NotificationAdoptionRefusedModel extends NotificationModel {
+  type: NotificationType.adoptionRefused,
+  animal: DocumentData,
+}
+
+export type NotificationModels = NotificationStandardModel | NotificationAdoptionModel | NotificationAdoptionRefusedModel;
+
+const sendToUser = (
+  user: DocumentRefData,
+  message: string,
+  type: NotificationType = NotificationType.standard,
+  data: (INotificationAdoptionInterestData | null) = null,
+): void => {
+  const currentUser = userAPI.currentUserDocument();
+  if (!currentUser) return;
+
+  firestore().collection('notifications').add({
+    from: currentUser,
+    to: user,
+    message,
+    seen: false,
+    type,
+    data,
```

Comment:
This seems wrong. The type definitions state that we should expect an "animal" key, not a "data" key with an animal object inside.

Reply:
Never mind, I just noticed the getModel function. A little unorthodox, but very effective way to get the animal data.